### DISCUSSION TITLE
Hacktoberfest 2021 for Everyone

### DISCUSSION THREAD

**Discussion by FeynmanZhou** [kubesphere/kubesphere#4322] [General]
## 👩‍💻 What is Hacktoberfest

[Hacktoberfest](https://hacktoberfest.digitalocean.com/) is a month-long celebration of open source projects run by [DigitalOcean](https://www.digitalocean.com/). Hacktoberfest is open to everyone, from first-time contributors to seasoned open source developers. [KubeSphere](https://github.com/kubesphere) is one of the projects that you can contribute to. 

## ☕️ Happy Hacking for KubeSphere Hacktoberfest 2021 

[KubeSphere](https://kubesphere.io/) team has collaborated with DigitalOcean to list KubeSphere to DigitalOcean Marketplace as a [one-click app](https://marketplace.digitalocean.com/apps/kubesphere) in July. We are so excited to be participating in Hacktoberfest 2021 as well. In the KubeSphere community, we believe that open source can help unlock the full potential of every developer. 

Swags, badges, and certifications are totally ready for new contributors. Are you ready to raise your first Pull Request to the KubeSphere community?

## 🙋‍♂️ How to Participate and Contribute

In general, there are two ways you can contribute to KubeSphere, both code and non-code contributions can be accepted by our community. 

It's recommended that you sign-up to Hacktoberfest on the [event website](https://hacktoberfest.digitalocean.com/) before you begin, then you can look through the issues marked as [hacktoberfest](https://github.com/search?q=user%3Akubesphere+user%3Akubesphere-sigs+label%3A%22hacktoberfest%22+state%3Aopen&type=Issues&ref=advsearch&l=&l=) to get you warmed up. Feel free to start with any open issues, unmarked ones included, or create new ones yourselves, leave your comments on those issues and self-assign. Just make sure to mention Hacktoberfest in your PR so that we can add the `hacktoberfest-accepted` label.

💡💡💡 We've curated a list of [beginner-friendly issues](https://github.com/search?q=user%3Akubesphere+user%3Akubesphere-sigs+label%3A%22hacktoberfest%22+state%3Aopen&type=Issues&ref=advsearch&l=&l=) for you to check out. We want everyone to start their open source journey with us.

#### Documentation

Improve, extend or add KubeSphere documentation on the official website.

#### i18n

Translate KubeSphere UI copywriting and landing page into different languages.

#### Advocacy and Outreach

Record videos, design artwork, write blogs, share your insight for newbies.

### Code

Help us to make a new feature come true, improve code, add test cases, or fix bugs.

## 🎁 Win Your Prizes

To qualify for the official limited Hacktoberfest 2021 T-shirt, KubeSphere swags, and certifications, you have to make four (4) pull requests (PRs) between October 1-31 (in any time zone). 

![image](https://user-images.githubusercontent.com/40452856/135841452-29b151be-0b36-4997-a17d-292b9250e425.png)

## ❤️ Sponsor Every Contributor

To help each contributor get started with KubeSphere and Kubernetes environment, we initiated a **Developer Support Program** to sponsor at least one VM with 8 Cores and 16 GB for you. You can apply the computing resources by sending an email to info@kubesphere.io.

**Comment by FeynmanZhou** [comment]
Hi team @kubesphere/developers , please help to create more GitHub issues labeled with `hacktoberfest` in your working repository, they will be included in the list of [beginner-friendly](https://github.com/search?q=user%3Akubesphere+user%3Akubesphere-sigs+label%3A%22hacktoberfest%22+state%3Aopen&type=Issues&ref=advsearch&l=&l=) issues for contributors to pick up.

**Comment by LinuxSuRen** [comment]
I just filed an issue to request `ks-installer` to add a label of hacktoberfest. See also https://github.com/kubesphere/ks-installer/issues/1747

**Comment by yongxingMa** [comment]
Hello，Is there any simple PR，I can try。

**Reply by FeynmanZhou** [reply]
Thanks for your joining! Which part are you interested in? Coding or non-code?

**Reply by yongxingMa** [reply]
> Thanks for your joining! Which part are you interested in? Coding or non-code?

none-code.

**Reply by FeynmanZhou** [reply]
@yongxingMa Could you take a look at this [issue](https://github.com/kubesphere/kubesphere/issues/4367)? It is a translation task that a new contributor could get on board.

**Comment by LinuxSuRen** [comment]
Contribution collection:

- @mangoGoForward
  - https://github.com/kubesphere-sigs/ks-devops-helm-chart/pull/30
  - https://github.com/kubesphere/ks-devops/pull/328
  - https://github.com/kubesphere/console/pull/2575
  - https://github.com/kubesphere/console/pull/2480
  - https://github.com/kubesphere-sigs/ks-devops-helm-chart/pull/39
  - https://github.com/kubesphere/helm-charts/pull/185

**Reply by mangoGoForward** [reply]
Another PR commited is
- kubesphere/console#2480

**Comment by LinuxSuRen** [comment]
I just created a request to invite @mangoGoForward as a new member of the KubeSphere community. I'm looking for a second sponsor. Please feel free to leave your comment in the following issue. Thanks!

See also https://github.com/kubesphere/community/issues/374
